src/cli/index.js

```diff
@@ -156,6 +162,46 @@ const getQuestions = ({ appName }) => ({
       when: ({ appId, apiKey, indexName }) =>
         !attributesToDisplay.length > 0 && appId && apiKey && indexName,
     },
+    {
+      type: 'checkbox',
+      name: 'attributesForFaceting',
+      message: 'Attributes to display',
+      suffix: `\n  ${chalk.gray('Used for filtering the search interface')}`,
+      pageSize: 10,
+      choices: async answers => {
+        const templatePath = getTemplatePath(answers.template);
+        const templateConfig = getAppTemplateConfig(templatePath);
+
+        const dynamicWidgets =
+          answers.libraryVersion &&
+          templateConfig.flags &&
+          templateConfig.flags.dynamicWidgets &&
+          semver.satisfies(
+            answers.libraryVersion,
+            templateConfig.flags.dynamicWidgets,
+            { includePrerelease: true }
+          )
```

Comment:
WDYT of moving this in a variable? That's a lot of inlining and makes the ternary a bit hard to read.

Reply:
like this?

```js
        const selectedLibraryVersion = answers.libraryVersion;
        const supportedLibraryVersion =
          templateConfig.flags && templateConfig.flags.dynamicWidgets;
        const canUseDynamicWidgets =
          selectedLibraryVersion &&
          supportedLibraryVersion &&
          semver.satisfies(selectedLibraryVersion, supportedLibraryVersion, {
            includePrerelease: true,
          });

        const dynamicWidgets = canUseDynamicWidgets
          ? [
              {
                name: 'Dynamic widgets',
                value: 'ais.dynamicWidgets',
                checked: true,
              },
              new inquirer.Separator(),
            ]
          : [];
```

Reply:
Yeah, or even simpler like:

```js
const supportsDynamicWidgets = /* ... */;
const dynamicWidgets = [/* ... */];

return [supportsDynamicWidgets && ...dynamicWidgets, /* ... */].filter(Boolean);
```

Reply:
Yeah, or even simpler like:

```js
const supportsDynamicWidgets = /* ... */;
const dynamicWidgets = [/* ... */];

return [supportsDynamicWidgets && ...dynamicWidgets, /* ... */].filter(Boolean);
```

Reply:
that would spread `false`, we need the conditional before spreading :)

Reply:
That's what the `filter(Boolean)` prevents.

Reply:
you can't spread after &&, that's not valid JS as far as I can tell :)

Reply:
Ah yep indeed, you'd have to `flat`, but let's probably not go there.

The point initially was just for a `supportsDynamicWidgets` variable so you can use it in your ternary.